lib/macho/headers.rb

```diff
@@ -61,17 +69,69 @@ module MachO
   # @see http://llvm.org/docs/doxygen/html/Support_2MachO_8h_source.html
   CPU_SUBTYPE_LIB64 = 0x80000000
 
-  # all x86-type CPUs
+  # all CPU sub-types for CPU type `CPU_TYPE_I386`
   CPU_SUBTYPE_X86_ALL = 3
 
-  # all x86-type CPUs (what makes this different from CPU_SUBTYPE_X86_ALL?)
-  CPU_SUBTYPE_X86_ARCH1 = 4
+  # all CPU sub-types for CPU type `CPU_TYPE_X86_64`
+  CPU_SUBTYPE_X86_64_ALL = CPU_SUBTYPE_X86_ALL
+
+  # all CPU sub-types for CPU type `CPU_TYPE_X86_64` (Haskell tree)
+  CPU_SUBTYPE_X86_64_H = 8
+
+  # all CPU sub-types for CPU type `CPU_TYPE_ARM`
+  CPU_SUBTYPE_ARM_V7 = 9
+  CPU_SUBTYPE_ARM_V7S = 11
+  CPU_SUBTYPE_ARM_V7K = 12
+
+  # all CPU sub-types for CPU-type `CPU_TYPE_ARM64`
+  CPU_SUBTYPE_ARM64_ALL = 0
+
+  # PowerPC G3
+  CPU_SUBTYPE_POWERPC_750 = 9
 
-  # association of cpu subtypes to string representations
+  # PowerPC G4
+  CPU_SUBTYPE_POWERPC_7400 = 10
+
+  # PowerPC G4 "Voyager"
+  CPU_SUBTYPE_POWERPC_7450 = 11
+
+  # PowerPC G5
+  CPU_SUBTYPE_POWERPC_970 = 100
+
+  # all CPU sub-types for CPU type `CPU_TYPE_POWERPC`
+  CPU_SUBTYPE_POWERPC_ALL = 0
```

Comment:
The comment above this doesn't really apply since you've added the corresponding sub-types in separate “sections” above this block. Not sure if they deserve this much room. Maybe put them all below this line and either omit the comment about the corresponding model or put that comment at the end of the line?